sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -94,25 +93,63 @@ class GpuWindowExpressionMeta(
             // queries because https://github.com/NVIDIA/spark-rapids/issues/1039 makes it so
             // we cannot support nulls in range queries
-            if (lower == Int.MinValue && upper == Int.MaxValue) {
+            // Will also verify that the types are what we expect.
+            if (spec.isUnbounded) {
               // this is okay because we will translate it to be a row query
             } else {
+              // check whether order by column is supported or not
               val orderSpec = wrapped.windowSpec.orderSpec
-              val allTime = orderSpec.forall { so =>
+              if (orderSpec.length > 1) {
+                // We only support a single time column
+                willNotWorkOnGpu("only a single date/time or integral (Boolean exclusive)" +
+                  "based column in window range functions is supported")
+              }
+              val orderByTypeSupported = orderSpec.forall { so =>
                 so.dataType match {
+                  case ByteType | ShortType | IntegerType | LongType => true
                   case DateType | TimestampType => true
                   case _ => false
                 }
               }
-              if (allTime) {
-                if (orderSpec.length > 1) {
-                  // We only support a single time column
-                  willNotWorkOnGpu("only a single date/time based column in window" +
-                      " range functions is supported")
+              if (!orderByTypeSupported) {
+                willNotWorkOnGpu(s"the type of orderBy column is not supported in a window" +
+                  s" range function, found ${orderSpec(0).dataType}")
+              }
+
+              def checkRangeBoundaryConfig(dt: DataType): Unit = {
+                dt match {
+                  case ByteType => if (!conf.isRangeWindowByteEnabled) willNotWorkOnGpu(
+                    s"Range window frame is not 100% compatible when the order by type is byte." +
+                      s" To enable it please set ${RapidsConf.ENABLE_RANGE_WINDOW_BYTES} to true.")
+                  case ShortType =>
+                    if (!conf.isRangeWindowShortEnabled) willNotWorkOnGpu(s"Range window frame" +
+                      s" is not 100% compatible when the order by type is short. To enable it" +
+                      " please set ${RapidsConf.ENABLE_RANGE_WINDOW_SHORT} to true.")
+                  case IntegerType =>
+                    if (!conf.isRangeWindowIntEnabled) willNotWorkOnGpu(
+                      s"Range window frame is not 100% compatible when the order by type is int." +
+                        s" To enable it please set ${RapidsConf.ENABLE_RANGE_WINDOW_INT} to true.")
+                  case LongType =>
+                    if (!conf.isRangeWindowLongEnabled) willNotWorkOnGpu(
+                      s"Range window frame is not 100% compatible when the order by type is long." +
```

Comment:
This isn't really a true statement, correct?

Reply:
it's true, when overflow even for long, the CPU and GPU will get different results